client/src/featureform/serving.py

```diff
@@ -15,122 +15,57 @@
+            self.impl = LocalClientImpl()
         else:
-            host = host or os.getenv('FEATUREFORM_HOST')
-            if host is None:
-                raise RuntimeError(
-                    'If not in local mode then `host` must be passed or the environment'
-                    ' variable FEATUREFORM_HOST must be set.'
-                )
-            if insecure:
-                channel = insecure_channel(host)
-            else:
-                channel = secure_channel(host, cert_path)
-            self._stub = serving_pb2_grpc.FeatureStub(channel)
+            self.impl = HostedClientImpl(host, insecure, cert_path)
 
 
     def training_set(self, name, version):
-        if self.local:
-            return self._local_training_set(name, version)
-        else:
-            return self._host_training_set(name, version)
-
-    def _host_training_set(self, name, version):
-        if self.local:
-            raise ValueError("Not supported in localmode. Please try using training_set()")
-        return Dataset(self._stub).from_stub(name, version)
+        return self.impl.training_set(name, version)
 
-    def _local_training_set(self, training_set_name, training_set_variant):
-        if not self.local:
-            raise ValueError("Only supported in localmode. Please try using dataset()")
-        training_set_row = \
-            self.sqldb.get_training_set_variant(training_set_name, training_set_variant)
-        label_row = \
-            self.sqldb.get_label_variant(training_set_row['label_name'], training_set_row['label_variant'])
-        label_source = self.sqldb.get_source_variant(label_row['source_name'], label_row['source_variant'])
-        if self.sqldb.is_transformation(label_row['source_name'], label_row['source_variant']):
-            df = self.process_transformation(label_row['source_name'], label_row['source_variant'])
-            if label_row['source_timestamp'] != "":
-                df = df[[label_row['source_entity'], label_row['source_value'], label_row['source_timestamp']]]
-                df[label_row['source_timestamp']] = pd.to_datetime(df[label_row['source_timestamp']])
-            else:
-                df = df[[label_row['source_entity'], label_row['source_value']]]
-            df.set_index(label_row['source_entity'])
-            label_df = df
+    def features(self, features, entities):
+        return self.impl.features(features, entities)
+
+class HostedClientImpl:
+    def __init__(self, host=None, insecure=False, cert_path=None):
+        host = host or os.getenv('FEATUREFORM_HOST')
+        if host is None:
+            raise ValueError(
+                'If not in local mode then `host` must be passed or the environment'
+                ' variable FEATUREFORM_HOST must be set.'
+            )
+        channel = self._create_channel(host, insecure, cert_path)
+        self._stub = serving_pb2_grpc.FeatureStub(channel)
+
+    def _create_channel(self, host, insecure, cert_path):
```

Comment:
Check tls.py in the src/featureform directory. I added functions to take care of this for all the clients so its more consistent

Reply:
Its being used in the resource client